src/Components/Aspire.Pomelo.EntityFrameworkCore.MySql/AspireEFMySqlExtensions.cs

```diff
@@ -75,11 +76,36 @@ void ConfigureDbContext(IServiceProvider serviceProvider, DbContextOptionsBuilde
 
             var connectionString = settings.ConnectionString ?? string.Empty;
 
-            ServerVersion serverVersion;
+            // Initialize serverVersion to a dummy value; the loop below will either replace it with an actual value or throw an unhandled exception.
+            ServerVersion serverVersion = null!;
+
             if (settings.ServerVersion is null)
             {
                 ConnectionStringValidation.ValidateConnectionString(settings.ConnectionString, connectionName, DefaultConfigSectionName, $"{DefaultConfigSectionName}:{typeof(TContext).Name}", isEfDesignTime: EF.IsDesignTime);
-                serverVersion = ServerVersion.AutoDetect(connectionString);
+
+                // Implement a simple backoff and retry loop based on MySqlRetryingExecutionStrategy (which has not been dependency-injected yet).
```

Comment:
Can we use Polly instead? See https://github.com/dotnet/aspire/issues/2077 for some links on how to use the new version.